File: BraintreeModels/src/test/java/com/braintreepayments/api/VenmoConfigurationUnitTest.java
package com.braintreepayments.api;

import org.json.JSONException;
import org.json.JSONObject;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertFalse;
import static junit.framework.Assert.assertTrue;

@RunWith(RobolectricTestRunner.class)
public class VenmoConfigurationUnitTest {

    @Test
    public void fromJson_parsesFullInput() throws JSONException {
        JSONObject input = new JSONObject()
                .put("accessToken", "sample-access-token")
                .put("environment", "sample-environment")
                .put("merchantId", "sample-merchant-id");

        VenmoConfiguration sut = VenmoConfiguration.fromJson(input);
        assertEquals("sample-access-token", sut.getAccessToken());
        assertEquals("sample-environment", sut.getEnvironment());
        assertEquals("sample-merchant-id", sut.getMerchantId());
        assertTrue(sut.isAccessTokenValid());
    }

    @Test
    public void fromJson_whenInputNull_returnsConfigWithDefaultValues() {
        VenmoConfiguration sut = VenmoConfiguration.fromJson(null);
        assertEquals("", sut.getAccessToken());
        assertEquals("", sut.getEnvironment());
        assertEquals("", sut.getMerchantId());
        assertFalse(sut.isAccessTokenValid());
    }

    @Test
    public void fromJson_whenInputEmpty_returnsConfigWithDefaultValues() {
        VenmoConfiguration sut = VenmoConfiguration.fromJson(new JSONObject());
        assertEquals("", sut.getAccessToken());
        assertEquals("", sut.getEnvironment());
        assertEquals("", sut.getMerchantId());
        assertFalse(sut.isAccessTokenValid());
    }
}

File: BraintreeCore/src/test/java/com/braintreepayments/api/VenmoConfigurationUnitTest.kt
package com.braintreepayments.api

import org.robolectric.RobolectricTestRunner
import org.json.JSONException
import org.json.JSONObject
import org.junit.Assert.*
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(RobolectricTestRunner::class)
class VenmoConfigurationUnitTest {
    @Test
    @Throws(JSONException::class)
    fun fromJson_parsesFullInput() {
        val input = JSONObject()
            .put("accessToken", "sample-access-token")
            .put("environment", "sample-environment")
            .put("merchantId", "sample-merchant-id")
        val sut = VenmoConfiguration(input)
        assertEquals("sample-access-token", sut.accessToken)
        assertEquals("sample-environment", sut.environment)
        assertEquals("sample-merchant-id", sut.merchantId)
        assertTrue(sut.isAccessTokenValid)
    }

    @Test
    fun fromJson_whenInputNull_returnsConfigWithDefaultValues() {
        val sut = VenmoConfiguration(null)
        assertEquals("", sut.accessToken)
        assertEquals("", sut.environment)
        assertEquals("", sut.merchantId)
        assertFalse(sut.isAccessTokenValid)
    }

    @Test
    fun fromJson_whenInputEmpty_returnsConfigWithDefaultValues() {
        val sut = VenmoConfiguration(JSONObject())
        assertEquals("", sut.accessToken)
        assertEquals("", sut.environment)
        assertEquals("", sut.merchantId)
        assertFalse(sut.isAccessTokenValid)
    }
}
